python/ray/serve/_private/replica_scheduler/pow_2_scheduler.py

```diff
@@ -453,23 +507,27 @@ async def choose_two_replicas_with_backoff(
                     self.num_scheduling_tasks_in_backoff
                 )
 
-    async def select_from_candidate_replicas(
+    async def _probe_queue_lens(
         self,
-        candidates: List[ReplicaWrapper],
+        replicas: List[ReplicaWrapper],
         backoff_index: int,
-    ) -> Optional[ReplicaWrapper]:
-        """Chooses the best replica from the list of candidates.
+    ) -> List[Tuple[ReplicaWrapper, Optional[int]]]:
+        """Actively probe the queue length from each of the replicas.
 
-        If none of the replicas can be scheduled, returns `None`.
+        Sends an RPC to each replica to fetch its queue length, with a response deadline
```

Comment:
Could you also mention that this function updates internal data structures to remove dead replicas?